### PR TITLE
String.equalsTo("") replaced with String.isEmpty() in core and tooling

### PR DESCRIPTION
**Pull request by dk2k**
String.equalsTo("") replaced with String.isEmpty(). The latter is more efficient. Separate commit for core and tooling
In MavenDownloaderImpl combined two similar IF blocks and removed useless assignment

### CODE REVIEW

File: tooling/camel-tooling-maven/src/main/java/org/apache/camel/tooling/maven/MavenDownloaderImpl.java [hunk 1]
```diff
@@ -731,10 +731,7 @@ Settings mavenConfiguration(
         // 3) ${user.home}/.m2/repository (if exists)
         // 4) /tmp/.m2/repository
         String localRepository = System.getProperty("maven.repo.local");
-        if (localRepository == null || "".equals(localRepository.trim())) {
-            localRepository = settings.getLocalRepository();
```
Comment (essobedo) [inline tooling/camel-tooling-maven/src/main/java/org/apache/camel/tooling/maven/MavenDownloaderImpl.java:735]: Why remove this?

Comment (dk2k) [reply tooling/camel-tooling-maven/src/main/java/org/apache/camel/tooling/maven/MavenDownloaderImpl.java:735]: The first step. Another IF with the same condition below. Both get combined into one block.
The second step. The assigned value isn't used before another value is assigned to localRepository => the firas assignment can be removed

Comment (essobedo) [reply tooling/camel-tooling-maven/src/main/java/org/apache/camel/tooling/maven/MavenDownloaderImpl.java:735]: Sorry I don't get it, what about the use case where `settings.getLocalRepository()` returns a non-empty value?

Comment (dk2k) [reply tooling/camel-tooling-maven/src/main/java/org/apache/camel/tooling/maven/MavenDownloaderImpl.java:735]: My bad. I understand now

**Review by essobedo — APPROVED** [review]
👍 LGTM, thx

**Review by davsclaus — APPROVED** [review]

**Pull request merged**